plugins/terraform/stepper.go

```diff
@@ -356,47 +356,87 @@ func GetBackendConfig(exec config.StepExecution, backendParser TFBackendParser)
 	// if user has decided to overwrite state file convention in backend.tf, support this override
 	if declaredBackend.Key != "" {
 		// grab statefile name (base)
-		b["key"] = interpolateString(exec, declaredBackend.Key)
+		interpolatedKey := interpolateString(exec, declaredBackend.Key)
+		b["key"] = interpolatedKey
 	}
 
-	if declaredBackend.S3RoleArn != "" {
-		b["role_arn"] = interpolateString(exec, declaredBackend.S3RoleArn)
+	if declaredBackend.AssumeRole.RoleArn != "" {
+		roleArn := declaredBackend.AssumeRole.RoleArn
+		exec.Logger.Tracef("Declared S3RoleArn: %s", roleArn)
+		interpolatedRoleArn := interpolateString(exec, declaredBackend.AssumeRole.RoleArn)
 
-		exec.Logger.Debugf("Declared S3RoleArn: %s", b["role_arn"])
+		// only track override config if interpolated is different from what user declared
+		if interpolatedRoleArn != roleArn {
+			roleArn = interpolatedRoleArn
+			b["assume_role"] = fmt.Sprintf("{\"role_arn\"=\"%s\"}", roleArn)
+		}
```

Comment:
if this suggestion is taken, then line 373 will log the wrong i.e. non-interpolated value. the suggestion is the way I originally coded it.

line 370 is what preserves the correct value outside the local scope, and that is being set correctly. so this is really just semantics of how to also log the chosen value (since we want the log entry regardless of the `if` condition on line 369).